src/photos.cs

```diff
@@ -400,9 +412,11 @@ interface PHAssetResourceRequestOptions : NSCopying
 	interface PHChange {
 
 		[Export ("changeDetailsForObject:")]
-		PHObjectChangeDetails GetObjectChangeDetails ([NullAllowed] PHObject obj);
```

Comment:
Is removing this `[NullAllowed]` attribute from the first method parameter a breaking change?

If I understand things correctly, this change will result in code being generated that throws a `NullArgumentException` when given null, and that wouldn't have happened until now. I tried to test passing `null` to this method before my change and didn't get an exception from Apple, but I'm also not sure if that's just relying on undocumented behavior.

Reply:
> Is removing this [NullAllowed] attribute from the first method parameter a breaking change?

It's a functional change but not a **breaking change** (as defined by .net). Nullability information is for
* macios generator (bgen) which add null-check **inside** methods; and
* csc (roslyn) compiler (for warnings)

This means binaries will still run, source will still build (might warn or error if `/warnaserror` is used, but that's fine).

> not sure if that's just relying on undocumented behavior.

The nullability information has not always been available from Apple. The old attributes were added based on headers comments, documentation, samples, unit tests and common sense.